src/pacemaker/protocol.rs

```diff
@@ -242,27 +287,46 @@ impl<N: Network> Pacemaker<N> {
         advance_view: AdvanceView,
         origin: &VerifyingKey,
         block_tree: &mut BlockTree<K>,
-    ) -> Result<(), UpdateViewError>{ 
-        if !block_tree.committed_validator_set().contains(origin) {return Ok(())};
+    ) -> Result<(), PacemakerError>{
+        Event::ReceiveAdvanceView(ReceiveAdvanceViewEvent{timestamp: SystemTime::now(), origin: origin.clone(), advance_view: advance_view.clone()})
+        .publish(&self.event_publisher);
+
+        let validator_set_state = block_tree.validator_set_state()?;
+        if !is_validator(origin, &validator_set_state) {return Ok(())}
```

Comment:
I think it is a bit odd that we do not process `AdvanceView` messages if we are not a validator, but we process `NewView` messages.

Reply:
No, we do process `AdvanceView` messages if we are not a validator, but we only process `AdvanceView` messages from validators (not from listeners) - this is what the line of code above means.

Reply:
But the comment above is confusing, I updated the comment in a new commit.